lib/thrift/parser/nimble.ex

```diff
@@ -0,0 +1,189 @@
+defmodule Thrift.Parser.Nimble do
+  import NimbleParsec
+
+  defmodule Lexer do
+    @moduledoc false
+    @punctuator ~c"(){}[]<>,;=*"
+
+    @keywords ~w(
+      namespace include cpp_include
+      typedef enum union struct exception
+      void bool byte i8 i16 i32 i64 double string binary list map set
+      const oneway extends throws service required optional
+      true false
+    )
+
+    def token(combinator \\ empty()) do
+      choice(combinator, [
+        keyword(),
+        identifier(),
+        literal(),
+        number(),
+        punctuator(),
+        whitespace()
+      ])
+    end
+
+    defp whitespace() do
+      ascii_string([?\s, ?\t, ?\n, ?\v, ?\f, ?\r], min: 1)
+      |> ignore()
+      |> label("whitespace")
+    end
+
+    defp keyword() do
+      @keywords
+      |> Enum.map(&(string(&1) |> replace(String.to_atom(&1))))
+      |> choice()
+      |> label("keyword")
+    end
+
+    defp punctuator() do
+      @punctuator
+      |> Enum.map(&(string(<<&1>>) |> replace(String.to_atom(<<&1>>))))
+      |> choice()
+      |> label("punctuator")
+    end
+
+    defp literal() do
+      choice([
+        literal_with(?"),
+        literal_with(?')
+      ])
+      |> reduce({List, :to_string, []})
+      |> label("literal")
+    end
+
+    defp literal_with(char) do
+      delim = ascii_char([char])
+
+      delim
+      |> ignore()
+      |> concat(
+        choice([
+          utf8_char([?\\]) |> ignore() |> concat(delim),
```

Comment:
Does this handle embedded newlines, etc. like we support in the current lexer?

https://github.com/pinterest/elixir-thrift/blob/1610f612e8638307e9b98970d35839299facf5f1/src/thrift_lexer.xrl#L91-L105